### ISSUE TITLE
Template editing: List existing template parts in the Inserter Patterns tab

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#41399]
## What problem does this address?
The Patterns inserter surfaces header and footer patterns that may have already been used to create template parts at the site. Consequently it's very easy to insert a _pattern_ in a template where an existing _template part_ might be preferable. 

## What is your proposed solution?
If we list existing template parts in the pattern inserter we can hopefully reduce the likelihood of such occurrences. We'd need to ensure the UI clearly distinguishes patterns / template parts. 

A couple of ideas to kick off discussions:

<img width="1177" alt="Screenshot 2022-05-27 at 12 14 37" src="https://user-images.githubusercontent.com/846565/170689238-80d557c3-2031-454b-9fe3-ccfc12ce3f1b.png">

**Comment by jameskoster** [comment]
For option C to work in the quick inserter we'd need to add the category dropdown there too. It could work, but in general I find scrolling and/or searching to be easier versus operating a dropdown. There's also a slightly awkward overlap... we'd have a "Template Parts" category, and a "Footers" category... it might be too confusing.

**Comment by ntsekouras** [comment]
@jameskoster do you think this issue is still relevant, given the inserter has been updated since the creation of it?

**Comment by jameskoster** [comment]
Yes I think so, particularly as concepts like Template Parts, Patterns, Reusable blocks continue to converge. 

It would be nice to find template parts here, ideally with ones that are in active use being visually marked somehow:

<img width="840" alt="Screenshot 2023-06-14 at 13 07 31" src="https://github.com/WordPress/gutenberg/assets/846565/bc58fb68-a303-4ca9-82ac-254d86cd99c6">
